src/driver.rs

```diff
@@ -131,19 +130,38 @@ pub fn block_on<T>(future: impl Future<Output = T>) -> T {
         let io_blocked = Arc::new(AtomicBool::new(false));
 
         // Prepare the waker.
-        let waker = waker_fn({
-            let io_blocked = io_blocked.clone();
-            move || {
-                if u.unpark() {
-                    // Check if waking from another thread and if currently blocked on I/O.
-                    if !IO_POLLING.with(Cell::get) && io_blocked.load(Ordering::SeqCst) {
-                        Reactor::get().notify();
-                    }
+        let waker = BlockOnWaker::create(io_blocked.clone(), u);
+
+        (p, waker, io_blocked)
+    }
+
+    struct BlockOnWaker {
+        io_blocked: Arc<AtomicBool>,
+        unparker: parking::Unparker,
+    }
```

Comment:
Could you move the structure and it's implementation after the `thread_local` declaration? Since it uses the thread local I think it will be less confusing for readers if it comes after.